andela/haven-ah-backend
# #161342840 Get and Update User Profile

#### What does this PR do?
- Create Get-and-update-user-profile functionality

#### Description of Task to be completed?
- Create the get user profile by username route
- Create the update user profile by username route
- Create utility for checking string length
- Create middleware to handle image upload and storage on cloudinary.com 

#### How should this be manually tested?
- git clone `https://github.com/andela/haven-ah-backend.git`.
- Install `node` if not installed.
- ensure that PostgreSQL is running on your local machine.
- run `npm install` to install the npm packages.
- run `npm test` to run the unit and integration tests.
To test on postman
- run `npm run start-dev`
- On postman run GET `localhost:5000/api/users/:username` (set JWT token as`x-access-token` in header).


#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#161342851

#### Screenshots (if appropriate)
Good Response:
<img width="1117" alt="screen shot 2018-11-01 at 2 07 31 pm" src="https://user-images.githubusercontent.com/31534129/47853627-9a3dff80-dddf-11e8-92a8-4bce6dabdcd4.png">


Bad Request Response (when email or username is included):
<img width="990" alt="screen shot 2018-10-31 at 10 48 12 am" src="https://user-images.githubusercontent.com/31534129/47779890-840a4380-dcfa-11e8-907e-b0c978046454.png">


When bio exceeds 200 character length:
<img width="990" alt="screen shot 2018-10-31 at 10 49 53 am" src="https://user-images.githubusercontent.com/31534129/47779990-c03da400-dcfa-11e8-8d1a-befb7c2ed53c.png">
